src/Microsoft.FeatureManagement/Targeting/ContextualTargetingFilter.cs

```diff
@@ -61,6 +61,27 @@ public Task<bool> EvaluateAsync(FeatureFilterEvaluationContext context, ITargeti
                 throw new ArgumentException(message, paramName);
             }
 
+            if (settings.Audience.Exclusion != null)
+            {
+                //
+                // Check if the user is in the exclusion directly
+                if (targetingContext.UserId != null &&
+                    settings.Audience.Exclusion.Users != null &&
+                    settings.Audience.Exclusion.Users.Any(user => targetingContext.UserId.Equals(user, ComparisonType)))
+                {
+                    return Task.FromResult(false);
+                }
+
+                //
+                // Check if the user is in a group within exclusion
+                if (targetingContext.Groups != null &&
+                    settings.Audience.Exclusion.Groups != null &&
+                    settings.Audience.Exclusion.Groups.Any(group => targetingContext.Groups.Contains(group)))
```

Comment:
group comparison needs the comparison type for case sensitivity.

Reply:
Updated. Had to do a nested `.Any( .Any(` since `Linq.Contains(` takes a `StringComparer` while the rest of the code uses `StringComparison`. Converting between the two isn't available in `netstandard2.0`, which we still have as a target.

Reply:
Ah, I see. I don't mind to add StringComparison as private member alongside StringComparer as well to avoid the nested `Any`, if you feel it's cleaner.

If you stick with nested Any though, I would like it if we broke up the line a bit since it's a bit long in length. Suggested

```
                    settings.Audience.Exclusion.Groups.Any(
                        group => targetingContext.Groups.Any(
                            targetingGroup => targetingGroup.Equals(group, ComparisonType))))


```